package tera;

enum StatusCode {
    // master
    kMasterNotInited = 12;
    kMasterIsBusy = 13;
    kMasterIsSecondary = 14;
    kMasterIsReadonly = 15;
    kMasterOnRestore = 16;
    kMasterOnWait = 17;
    kMasterIsRunning = 19;

    // tablet node
    kTabletNodeNotInited = 22;
    kTabletNodeIsBusy = 23;
    kTabletNodeIsIniting = 24;
    kTabletNodeIsReadonly = 25;
    kTabletNodeIsRunning = 29;

    // tablet node manage
    kTabletNodeReady = 30;
    kTabletNodeOffline = 31;
    kTabletNodeOnKick = 32;
    kTabletNodeWaitKick = 33;
    kTabletNodePendingOffline = 34;
    kTabletNodeKicked   = 35;
    // response
    kTabletNodeOk = 40;

    // table status
    kTabletNotInit = 41;
    kTabletReady = 42;
    kTabletOnLoad = 43;
    kTabletOnSplit = 44;
    kTabletMerged = 50;
    kTabletSplitted = 51;
    kTabletUnloading = 52;
    kTabletUnloading2 = 68;
    kTabletWaitLoad  = 69;
    kTabletWaitUnload = 70;


    kTableNotFound = 45;
    kTableCorrupt = 46;
    kTableNotSupport = 47;
    kTableInvalidArg = 48;

    // ACL & system
    kIllegalAccess = 71;
    kNotPermission = 72;
    kIOError = 73;
    kMismatchAuthType = 74;

    kTxnFail = 80;

    // quota
    kQuotaLimited = 90;
    kQuotaInvalidArg = 91;
    kFlowControlLimited = 92;

    //// master rpc ////

    // create&update table
    kMasterOk = 200;
    kTableExist = 201;
    kTableNotExist = 202;
    kTableIsBusy = 203;
    kTableMergeError = 204;

    // register
    kInvalidSequenceId = 304;
    kInvalidTabletNodeInfo = 305;

    // report
    kTabletNodeNotRegistered = 306;

    // cmdctrl
    kInvalidArgument = 310;

    // snapshot
    kSnapshotNotExist = 320;

    //// tablet node rpc ////

    // key
    kKeyNotExist = 402;
    kKeyNotInRange = 403;

    // meta table
    kMetaTabletError = 500;

    // RPC
    kRPCError = 700;
    kServerError = 701;
    kClientError = 702;
    kConnectError = 703;
    kRPCTimeout = 704;

    // Async Writer
    kAsyncNotRunning = 801;
    kAsyncTooBusy = 802;
    kAsyncFailure = 803;

    // zk
    kZKError = 900;

    kTableStatusEnable = 1000;
    kTableStatusDisable = 1001;

    // Timeoracle
    kTimeoracleOk   = 2000;
    kTimeoracleBusy = 2001;

    // rowlock service
    kLockSucc = 2100;
    kLockFail = 2101;
    kLockUnknown = 2102;

    // LoadBalancer
    kLoadBalancerOk = 2200;
    kLoadBalancerError = 2201;
}

enum TableStatus {
    kTableEnable = 1000;
    kTableDisable = 1001;
    kTableDeleting = 1002;
}

enum CompactStatus {
    kTableNotCompact = 54;
    kTableOnCompact = 55;
    kTableCompacted = 56;
}
